src/ResourceManager/KeyVault/Commands.KeyVault/Commands/GetAzureKeyVaultCertificate.cs

```diff
@@ -88,56 +138,71 @@ public class GetAzureKeyVaultCertificate : KeyVaultCmdletBase
+                   HelpMessage = "Specifies whether to show the previously deleted certificates in the output." )]
+        [Parameter(Mandatory = false,
+                   ParameterSetName = InputObjectByVaultNameParameterSet,
+                   HelpMessage = "Specifies whether to show the previously deleted certificates in the output.")]
         public SwitchParameter InRemovedState { get; set; }
         #endregion
 
-        protected override void ProcessRecord()
+        public override void ExecuteCmdlet()
         {
             CertificateBundle certBundle;
 
-            switch (ParameterSetName)
+            if (InputObject != null)
             {
-                case ByCertificateNameParameterSet:
-                    certBundle = this.DataServiceClient.GetCertificate(VaultName, Name, Version ?? string.Empty);
-                    var certificate = KeyVaultCertificate.FromCertificateBundle(certBundle);
-                    this.WriteObject(certificate);
-                    break;
+                VaultName = InputObject.VaultName.ToString();
+            }
 
-                case ByCertificateVersionsParameterSet:
-                    certBundle = this.DataServiceClient.GetCertificate(VaultName, Name, string.Empty);
-                    if (certBundle != null)
-                    {
-                        WriteObject(new CertificateIdentityItem(certBundle));
-                        GetAndWriteCertificatesVersions(VaultName, Name, certBundle.CertificateIdentifier.Version);
-                    }
-                    break;
-
-                case ByVaultNameParameterSet:
+            if (!string.IsNullOrEmpty(Version))
+            {
+                certBundle = this.DataServiceClient.GetCertificate(VaultName, Name, Version);
+                var certificate = PSKeyVaultCertificate.FromCertificateBundle(certBundle);
+                this.WriteObject(certificate);
+            }
+            else if (IncludeVersions)
+            {
+                certBundle = this.DataServiceClient.GetCertificate(VaultName, Name, string.Empty);
+                if (certBundle != null)
+                {
+                    WriteObject(new PSKeyVaultCertificateIdentityItem(certBundle));
+                    GetAndWriteCertificatesVersions(VaultName, Name, certBundle.CertificateIdentifier.Version);
+                }
+            }
+            else if (InRemovedState)
+            {
+                if (Name == null)
+                {
+                    GetAndWriteDeletedCertificates(VaultName);
+                }
+                else
+                {
+                    PSDeletedKeyVaultCertificate deletedCert = PSDeletedKeyVaultCertificate.FromDeletedCertificateBundle(DataServiceClient.GetDeletedCertificate(VaultName, Name));
+                    WriteObject(deletedCert);
+                }
+            }
+            else
+            {
+                if (string.IsNullOrEmpty(Name))
```

Comment:
It's the structure of the 'if' block that looks odd - would it be possible to refactor this with less nesting?